esphome/components/uart/uart_component_esp_idf.cpp

```diff
@@ -123,11 +123,19 @@ void IDFUARTComponent::load_settings(bool dump_config) {
     return;
   }
 
-  if (this->rx_pin_) {
-    this->rx_pin_->setup();
-  }
-  if (this->tx_pin_ && this->rx_pin_ != this->tx_pin_) {
-    this->tx_pin_->setup();
+  auto setup_pin_if_needed = [](InternalGPIOPin *pin) {
+    if (!pin) {
+      return;
+    }
+    const auto mask = gpio::Flags::FLAG_OPEN_DRAIN | gpio::Flags::FLAG_PULLUP | gpio::Flags::FLAG_PULLDOWN;
+    if (pin->get_flags() & mask) {
```

Comment:
```suggestion
    if ((pin->get_flags() & mask) != gpio::Flags::FLAG_NONE) {
```

libretiny version is a bit more explict and clearer about the intent